typescript/lib/parsers/palm.ts

```diff
@@ -0,0 +1,50 @@
+import { DiscussServiceClient } from "@google-ai/generativelanguage";
+import { JSONObject } from "../../common";
+import { Prompt, Output } from "../../types";
+import { AIConfigRuntime } from "../config";
+import { InferenceOptions } from "../modelParser";
+import { ParameterizedModelParser } from "../parameterizedModelParser";
+import { GoogleAuth } from "google-auth-library";
+import { getAPIKeyFromEnv } from "../utils";
+
+/**
+ * Model Parser for PaLM Text Generation. PaLM API Currently does not support streaming
+ * models/chat-bison-001
+ */
+
+export class PaLMTextParser extends ParameterizedModelParser {
+  _id = "models/text-bison-001";
+
+  public serialize(promptName: string, data: JSONObject, aiConfig: AIConfigRuntime, params?: JSONObject | undefined): Prompt | Prompt[] {
```

Comment:
Didn't we want only to return array type instead of singular type `Prompt[]`? We didn't fix this during Eng Week? Let's create a TODO + link to the Github Issues. Actually yea, let's do all TODOs linked to GH issues. I'll do this too

Reply:
Yes thanks for catching that, looks like I had an unpushed change, updated. Also created #416